packages/ListBox/src/store/Provider.js

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import PropTypes from "prop-types";
+import reducer from "./reducer";
+import { getDataOptions } from "../components/Option/helpers/optionState";
+
+export const StoreContext = React.createContext();
+
+function initializeState(props) {
+  const {
+    childrenOptions,
+    height,
+    isDisabled,
+    isInline,
+    isMulti,
+    isOpen,
+    onChange,
+    placeholder,
+    refFilterInput,
+    refFooterContainer,
+    refListBox,
+    refListBoxContainer,
+    refTrigger,
+    refTriggerContainer,
+  } = props;
+
+  // initialize state for options and groups
+  const options = getDataOptions(childrenOptions);
+  // const Footer = getFooter(childrenOptions);
+  const selectedOptions = Object.keys(options)
+    .filter(key => options[key].isSelected)
+    .map(key => Number.parseInt(key, 10));
+
+  let activeOption = null;
+  if (selectedOptions.length && !props.isMulti) {
+    activeOption = selectedOptions[0];
+  }
+
+  const initialState = {
+    activeOption,
+    filteredOptions: [],
+    hasFooter: false,
+    hasPopupOpened: false,
+    height,
+    isDisabled,
+    isInline,
+    isMulti,
+    isOpen,
+    noResultsFound: false,
+    onChange,
+    options,
+    placeholder,
+    refFilterInput,
+    refFooterContainer,
+    refListBox,
+    refListBoxContainer,
+    refTrigger,
+    refTriggerContainer,
+    selectedOptions,
+    shouldContentScroll: true,
+    triggerWidth: 0,
+  };
+
+  return {
+    ...initialState,
+    originalState: { ...initialState },
+    lastKnownSelectedOptions: initialState.selectedOptions.slice(0),
+  };
+}
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+export default function Provider(props) {
+  const refFilterInput = React.useRef(null);
+  const refFooterContainer = React.useRef(null);
+  const refListBox = React.useRef(null);
+  const refListBoxContainer = React.useRef(null);
+  const refTrigger = React.useRef(null);
+  const refTriggerContainer = React.useRef(null);
+
+  const [state, dispatch] = React.useReducer(
+    reducer,
+    {
+      ...props,
+      refFilterInput,
+      refFooterContainer,
+      refListBox,
+      refListBoxContainer,
+      refTrigger,
+      refTriggerContainer,
+    },
+    initializeState
+  );
+
+  const value = { state, dispatch };
```

Comment:
Actually, I'm recalling now that the recommended pattern is to have two different contexts for state and dispatch: https://reactjs.org/docs/hooks-faq.html#how-to-avoid-passing-callbacks-down
Then you only need to memoize the start part because dispatch is guaranteed to be stable.

Reply:
Sound to me like a major refactor having two contexts for this PR.  

I have a list of improvements for this component. Should I add this suggestion to it? 

(Note: I'm down to do it but don't know if this PR is the best option)